File: 프로그래머스/2/42586. 기능개발/기능개발.js
/**
각 배포는 순서대로 진행되어야함 (뒤에꺼가 먼저 완성되어도 먼저 배포 불가능)

구하는 것 : 각 배포마다 몇 개의 기능이 배포되는가

*/
function solution(progresses, speeds) {
    const answer = [];
    
    while(progresses.length > 0){
        let count = 0;
        for(let i = 0 ; i < progresses.length; i++){
            if(progresses[i] < 100){
                progresses[i] += speeds[i];
            }
        }
        
        while(true){
            if(progresses[0] >= 100){
                progresses.shift();
                speeds.shift();
                count++;
            }else{
                break;
            }
        }
        
        if(count > 0) answer.push(count);
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    end_day = []
    
    # task별 종료까지 걸리는 날짜 계산
    for idx, task in enumerate(progresses):
        end_day.append(math.ceil((100-task) / speeds[idx]))
    
    task_cnt = 1
    cur = end_day[0]
    
    for i in range(1, len(end_day)):
        if(cur >= end_day[i]):
            task_cnt += 1
        else:
            answer.append(task_cnt)
            task_cnt = 1
            cur = end_day[i]
            
    answer.append(task_cnt)
    
    return answer